tcms/kiwi_auth/test_admin.py

```diff
@@ -109,3 +111,57 @@ def test_admin_can_add_new_users(self):
 
         self.assertEqual(HTTPStatus.OK, response.status_code)
         self.assertTrue(get_user_model().objects.filter(username='added-by-admin').exists())
+
+
+class TestGroupAdmin(LoggedInTestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.admin = UserFactory()
+        cls.admin.is_superuser = True
+        cls.admin.set_password('admin-password')
+        cls.admin.save()
+
+        cls.group = GroupFactory(name='NewGroupName')
+        cls.group.save()
+        cls.defaultGroups = Group.objects.filter(name__in=['Administrator', 'Tester'])
+
+    def setUp(self):
+        super().setUp()
+        self.client.login(  # nosec:B106:hardcoded_password_funcarg
+            username=self.admin.username,
+            password='admin-password')
+
+    def test_should_not_be_allowed_to_change_default_groups_names(self):
+        for group in self.defaultGroups:
+            response = self.client.get(reverse('admin:auth_group_change', args=[group.id]))
+            self.assertNotContains(response,
+                                   '<input type="text" name="name" value="%s" class="vTextField"'
+                                   ' maxlength="150" required="" id="id_name">'
+                                   % group.name)
+            self.assertContains(response, '<div class="grp-readonly">%s</div>' % group.name)
+
+    def test_should_not_be_allowed_to_delete_default_groups_names(self):
+        for group in self.defaultGroups:
+            response = self.client.get(reverse('admin:auth_group_change', args=[group.id]))
+            self.assertNotContains(response,
+                                   '<input type="text" name="name" value="%s" class="vTextField"'
+                                   ' maxlength="150" required="" id="id_name">'
+                                   % group.name)
+            self.assertNotContains(response,
+                                   '<a href="%s" class="grp-button grp-delete-link">Delete</a>'
+                                   % reverse('admin:auth_group_delete', args=[self.group.id]))
+
+    def test_should_be_allowed_to_create_new_group(self):
```

Comment:
This scenario is incorrect.  All of the changes in the GroupAdmin class will influence how the page is displayed and what fields are there, not how a POST request is handled.

Hence for this scenario you should load the "add new group" page and make sure it allows the user to enter a group name, very similar to how you can edit a non-default group.

Everything that happens after the Save button is clicked is handled internally by Django and at this moment we don't care to test it b/c that's Django's responsibility.